torch/_refs/__init__.py

```diff
@@ -1539,6 +1539,24 @@ def sub(
 )
 
 
+@register_decomposition(torch.ops.aten.xlogy)
+@out_wrapper()
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a", "b"),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
+)
+def xlogy(a: Union[TensorLikeType, NumberType], b: Union[TensorLikeType, NumberType]):
+    assert isinstance(a, TensorLike) or isinstance(b, TensorLike)
+    if isinstance(a, TensorLike) and isinstance(b, Number):
```

Comment:
This is an awkward construction because it constructs the `scalar_tensor` in the datatype of the tensor, but it's OK to do so because the type promotion decorator has already converted the tensor to the appropriate dtype

Reply:
These conditions can be simplified since one of `a` and `b` is a tensorlike object per the above check:

```
a = torch.scalar_tensor(a, b.dtype, b.device) if isinstance(a, Number) else a
b = torch.scalar_tensor(b, a.dtype, a.device) if isinstance(a, Number) else b
```

Reply:
Darn. I really don't like mypy